src/org/exist/indexing/StreamListener.java

```diff
@@ -43,6 +43,13 @@
         STORE,
 
         /**
+         * Mode for replacing the nodes of a document with another document
+         * this is really a group mode, which is later followed by {@link #REMOVE_ALL_NODES}
+         * and then {@link #STORE}
+         */
+        REPLACE_DOCUMENT,
```

Comment:
And last question - how critical to have `REPLACE_DOCUMENT`? Will you do same by having only `startReplaceDocument` and `endReplaceDocument` methods?